lib/paperclip/attachment.rb

```diff
@@ -613,5 +614,21 @@ def able_to_store_created_at?
     def has_enabled_but_unset_created_at?
       able_to_store_created_at? && !instance_read(:created_at)
     end
+
+    # Checks whether the option to show attributes to be destroyed is enabled and whether a destroy callback has been
+    # invoked when deleting the entire record.
+    def show_attrs_and_destroy_callback_triggered?
+      @options[:return_file_attributes_on_destroy] && instance.instance_variable_get(:@_destroy_callback_already_called)
+    end
+
+    # Sets attachment-related attributes to `nil`.
+    def clear_all_attachment_attributes
+      instance_write(:file_name, nil)
+      instance_write(:content_type, nil)
+      instance_write(:file_size, nil)
+      instance_write(:fingerprint, nil)
+      instance_write(:created_at, nil) if has_enabled_but_unset_created_at?
+      instance_write(:updated_at, nil)
```

Comment:
Also, any clue why were the attributes being set to `nil`? It seems to have been added during very early days of the project.

https://github.com/kreeti/kt-paperclip/commit/5494e3cfb6f0a85fb41ee3bead39fc269dbfd17e#diff-a00bc7cd3a1cbcb5ea7b3ab1c666d8c9caaba8907c1c1fe9327e41afef3d4dd6R85

Reply:
Well, I have no idea, to be honest. Maybe we should get rid of it, WDYT?